DeusEx/Classes/DXRReduceItems.uc

```diff
@@ -1,20 +1,119 @@
+};
+struct sMaxAmmo {
+    var class<Ammo> type;
+    var int percent;
+};
+
+var config int mission_scaling[16];
+var config sReduceAmmo ammo_reductions[16];
+var config sReduceItem reduce_items[16];
+var config sSetMax max_copies[16];
+var config sMaxAmmo max_ammo[16];
+
+function CheckConfig()
+{
+    local int i;
+    if( config_version == 0 ) {
+        for(i=0; i < ArrayCount(mission_scaling); i++) {
+            mission_scaling[i] = 100;
+        }
+        for(i=0; i < ArrayCount(ammo_reductions); i++) {
+            ammo_reductions[i].type = None;
+        }
+        for(i=0; i < ArrayCount(reduce_items); i++) {
+            reduce_items[i].type = None;
+        }
+        for(i=0; i < ArrayCount(max_copies); i++) {
+            max_copies[i].type = None;
+        }
+        for(i=0; i < ArrayCount(max_ammo); i++) {
+            max_ammo[i].type = None;
+        }
+    }
+    Super.CheckConfig();
+}
+
 function FirstEntry()
 {
+    local int i, mission, scale;
     Super.FirstEntry();
 
-    ReduceAmmo( float(dxr.flags.ammo)/100.0 );
-    ReduceSpawns(class'Multitool', dxr.flags.multitools);
-    ReduceSpawns(class'Lockpick', dxr.flags.lockpicks);
-    ReduceSpawns(class'BioelectricCell', dxr.flags.biocells);
-    ReduceSpawns(class'MedKit', dxr.flags.medkits);
+    mission = Clamp(dxr.dxInfo.missionNumber, 0, ArrayCount(mission_scaling)-1);
+    scale = mission_scaling[mission];
+
+    ReduceAmmo(class'Ammo', float(dxr.flags.ammo*scale)/100.0/100.0);
+    SetMaxAmmo( class'Ammo', dxr.flags.ammo );
+
+    ReduceSpawns(class'Multitool', dxr.flags.multitools*scale/100);
+    ReduceSpawns(class'Lockpick', dxr.flags.lockpicks*scale/100);
+    ReduceSpawns(class'BioelectricCell', dxr.flags.biocells*scale/100);
+    ReduceSpawns(class'MedKit', dxr.flags.medkits*scale/100);
+
+    SetMaxCopies(class'Multitool', dxr.flags.multitools*scale/100);
+    SetMaxCopies(class'Lockpick', dxr.flags.lockpicks*scale/100);
+    SetMaxCopies(class'BioelectricCell', dxr.flags.biocells*scale/100);
+    SetMaxCopies(class'MedKit', dxr.flags.medkits*scale/100);
+
+    for(i=0; i < ArrayCount(ammo_reductions); i++) {
+        if( ammo_reductions[i].type == None ) continue;
+        ReduceAmmo(ammo_reductions[i].type, float(ammo_reductions[i].percent*scale)/100.0/100.0 );
+    }
+    for(i=0; i < ArrayCount(reduce_items); i++) {
+        if( reduce_items[i].type == None ) continue;
+        ReduceSpawns(reduce_items[i].type, reduce_items[i].percent*scale/100 );
+    }
+    for(i=0; i < ArrayCount(max_copies); i++) {
+        if( max_copies[i].type == None ) continue;
+        SetMaxCopies( max_copies[i].type, max_copies[i].percent*scale/100 );
+    }
+    for(i=0; i < ArrayCount(max_ammo); i++) {
+        if( max_ammo[i].type == None ) continue;
+        SetMaxAmmo( max_ammo[i].type, max_ammo[i].percent*scale/100 );
+    }
+    SetTimer(1.0, true);
 }
 
-function ReduceAmmo(float mult)
+function Timer()
+{
+    local int i, mission, scale;
+    Super.Timer();
+
```

Comment:
needs to run on a timer because when the player picks up an item, I think it gets reset back to the default maxCopies